File: src/main/kotlin/pl/ue/poznan/matriculation/irk/domain/applications/Application.kt
package pl.ue.poznan.matriculation.irk.domain.applications


import com.fasterxml.jackson.annotation.JsonProperty
import pl.ue.poznan.matriculation.irk.dto.applications.ForeignerDataDTO
import pl.ue.poznan.matriculation.local.domain.Turn
import pl.ue.poznan.matriculation.local.domain.applicants.Applicant

data class Application(
        val admitted: String?,
        val comment: String?,
        @JsonProperty("foreigner_data")
    val foreignerDataDTO: ForeignerDataDTO?,
        val id: Long,
        val payment: String?,
        val position: String?,
        val qualified: String?,
        val score: String?,
        val turnDTO: Turn?,
        var user: Applicant?
)

File: src/main/webapp/src/app/application.ts
export interface Application {
    id: number,
            user: number,
            payment: string,
            score: string,
            position: string,
            qualified: string,
            admitted: string,
            comment: string,
            turn: {
                programme: string,
                registration: string,
                date_from: Date,
                date_to: Date
            },
            foreigner_data: {
                base_of_stay: string,
                source_of_financing: string,
                basis_of_admission: string
            }
}
